_drafts/gdbundle.md

```diff
@@ -0,0 +1,526 @@
+---
+title: "gdbundle - GDB's Missing Plugin Manager"
+description: "Introducing gdbundle, a plugin manager for GDB and LLDB, which enables developers to easily install plugins through PyPi and pip."
+author: tyler
+image:
+---
+
+I started with embedded development at Pebble, the smart watch maker, where we
+were building a modern operating system on a device with 128 KB of RAM. It's
+also where I got up-close and personal with GDB and had my first encounter with
+GDB's Python API.
+
+One one of our bi-weekly hack-days, one of the firmware engineers wrote a script
+to pretty-print a summary of the heap for a device connected to GDB using its
+Python API ([similar to
+this]({% post_url 2019-07-02-automate-debugging-with-gdb-python-api %}#adding-a-custom-gdb-command-with-gdb-python)).
+It was **awesome**. But, using this script with GDB wasn't without problems. We
+slowly grew it into a collection of scripts and improved our workflows. A few
+notable improvements were:
+
+- Add more scripts for each of our critical modules in the firmware, such as
+  state machines, Bluetooth state, etc.
+- Instead of just printing the heap allocations, expose the heap through a
+  Python Class. e.g.
+  `heap = Heap(g_heap); total_size = sum([b.size] for b in heap.iter_blocks()])`.
+- [Wrapping the GDB
+  invocation]({% post_url 2019-08-27-building-a-cli-for-firmware-projects %})
+  and loading GDB Python scripts automatically so every developer on the team
+  had them loaded and ready to use at all times.
+
+This was all back in 2015 or so. As I've moved around to other companies and
+talked to many other embedded software developers, I've discovered that many of
```

Comment:
Maybe: go with something positive rather than something negative. Like:

"... moved to other companies, I've had the joy to spread the gospel of GDB scripting to other embedded software developers. It's amazing how much a few python scripts can improve one's day to day work".